components/server/src/code-sync/code-sync-service.ts

```diff
@@ -253,24 +252,10 @@ export class CodeSyncService {
                 if (response.status !== 200) {
                     throw new Error(`code sync: blob service: upload failed with ${response.status} ${response.statusText}`);
                 }
-                oldObject = oldRev && toObjectName(resourceKey, oldRev);
+                oldRevList = oldRevs
             }, { latestRev, revLimit });
-            if (oldObject) {
-                const request = new DeleteRequest();
-                request.setOwnerId(userId);
-                request.setExact(oldObject);
-
-                const blobsClient = this.blobsProvider.getDefault();
-                blobsClient.delete(request, (err: ServiceError | null) => {
-                    if (err) {
-                        if (err.code === status.NOT_FOUND) {
-                            // we're good here
-                            return;
-                        }
-                        log.error({ userId }, 'code sync: failed to delete', err, { object: oldObject });
-                    }
-                });
-            }
+            // sync delete old revs from storage
+            this.deleteObjects(userId, resourceKey, oldRevList)
```

Comment:
@mustard-mh I just found we have lint check [server package.json](https://github.com/gitpod-io/gitpod/blob/a81c23ee504048414865a53de7af81d2a33b21cf/components/server/package.json#L20-L21)
If you run it we get an errorin this line `ERROR: 258:13  no-floating-promises  Promises must be handled appropriately` so maybe we should go with https://github.com/gitpod-io/gitpod/pull/8644#discussion_r822938267 ?

Reply:
I think we can just pass `.catch(e => {})` to it to pass this lint

Reply:
I think that's kinda misleading but your call :+1: